data/post-router.js

```diff
@@ -0,0 +1,106 @@
+const express = require("express");
+
+const Posts = require(`./db.js`);
+
+const router = express.Router();
+
+//GET ALL POSTS
+
+router.get("/", async (req, res) => {
+  try {
+    const posts = await Posts.find(req.query);
+    res.status(200).json(posts);
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({
+      error: "The posts information could not be retrieved."
+    });
+  }
+});
+
+//GET SINGLE POST
+
+router.get("/:id", async (req, res) => {
+  try {
+    const post = await Posts.findById(req.params.id);
+    console.log(post);
+    if (post[0]) {
+      res.status(200).json(post);
+    } else {
+      res
+        .status(404)
+        .json({ message: "The post with the specified ID does not exist." });
+    }
+  } catch (error) {
+    res.status(500).json({
+      error: "The post information could not be retrieved."
+    });
+  }
+});
+
+//POST
+
+router.post("/", async (req, res) => {
+  try {
+    const title = await req.body.title;
+    const contents = await req.body.contents;
+    if (title && contents) {
+      res.status(201).json(req.body);
```

Comment:
You forgot to call the insert() function. 

Insert takes in one argument, a post object. It returns an object that only contains the newly created post's id. We can capture that returned object with a falcon. If falcon is not undefined, that means our post was a success. 

```suggestion
router.post("/", async (req, res) => {
  try {
    const falcon = await Posts.insert(req.body); 
    if (falcon) {
      res.status(201).json(falcon);
```